src/main/java/cloud/fogbow/ras/core/OrderController.java

```diff
@@ -119,13 +119,62 @@ public void closeOrder(Order order) throws InternalServerErrorException {
 
     public void pauseCompute(Order order) throws FogbowException {
         synchronized (order) {
+            OrderState orderState = order.getOrderState();
+            if (order.isRequesterLocal(this.localProviderId) && hasOrderDependencies(order.getId())) {
+                throw new UnacceptableOperationException(String.format(Messages.Exception.DEPENDENCY_DETECTED_S_S,
+                        order.getId(), this.orderDependencies.get(order.getId())));
 
+            } else if (orderState.equals(OrderState.PAUSED)) {
+                throw new UnacceptableOperationException(Messages.Exception.VIRTUAL_MACHINE_ALREADY_PAUSED);
+
+            } else if(orderState.equals(OrderState.PAUSING)) {
+                throw new UnacceptableOperationException(Messages.Exception.PAUSE_OPERATION_ONGOING);
+
+            } else if (!orderState.equals(OrderState.FULFILLED)) {
+                throw new UnacceptableOperationException(Messages.Exception.VIRTUAL_MACHINE_IS_NOT_RUNNING);
+
+            } else if (order.isProviderRemote(this.localProviderId)) {
+                try {
+                    RemoteCloudConnector remoteCloudConnector = (RemoteCloudConnector)
+                            CloudConnectorFactory.getInstance().getCloudConnector(order.getProvider(), order.getCloudName());
+                    remoteCloudConnector.pauselInstance(order);
+                    OrderStateTransitioner.transitionToRemoteList(order, OrderState.PAUSING);
+                } catch (Exception e) {
+                    LOGGER.error(Messages.Exception.UNABLE_TO_RETRIEVE_RESPONSE_FROM_PROVIDER_S);
+                    throw e;
+                }
+            } else {
+                OrderStateTransitioner.transition(order, OrderState.PAUSING);
```

Comment:
I think you should call the LocalCloudConnector pauseCompute method that will call the plugin. In fcat, the code can be simplified by using the CloudConnectorFactory to create the appropriate CloudConnector object (local or remote), and then call the CloudConnector pauseCompute method (without the need to test whether the request is local or remote).